File: app/src/main/java/com/yao/zhihudaily/ui/daily/CommentsActivity.java
package com.yao.zhihudaily.ui.daily;

import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;
import com.yao.zhihudaily.R;
import com.yao.zhihudaily.model.StoryExtra;
import com.yao.zhihudaily.net.UrlConstants;
import com.yao.zhihudaily.ui.BaseActivity;

import java.util.ArrayList;

import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 *
 * @author Administrator
 * @date 2016/8/30
 */
public class CommentsActivity extends BaseActivity {

    @BindView(R.id.tab_layout)
    TabLayout mTabLayout;
    @BindView(R.id.view_pager)
    ViewPager mViewPager;
    @BindView(R.id.toolbar)
    Toolbar mToolbar;

    private StoryExtra mStoryExtra;
    private ArrayList<Fragment> mFragmentArrayList = new ArrayList<>(2);


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_comments);
        ButterKnife.bind(this);

        int id = getIntent().getIntExtra("id", 0);
        mStoryExtra = (StoryExtra) getIntent().getSerializableExtra("mStoryExtra");

        initToolbar(mToolbar);


        ArrayList<String> tabList = new ArrayList<>();
        tabList.add("短评论(" + mStoryExtra.getShortComments() + ")");
        tabList.add("长评论(" + mStoryExtra.getLongComments() + ")");
        mTabLayout.addTab(mTabLayout.newTab().setText(tabList.get(0)));//添加tab选项卡
        mTabLayout.addTab(mTabLayout.newTab().setText(tabList.get(1)));

        //短评论界面
        CommentsFragment shortCommentsFragment = new CommentsFragment();
        Bundle bundleForShortComments = new Bundle();
        bundleForShortComments.putInt("id", id);
        bundleForShortComments.putString("url", UrlConstants.SHORT_COMMENTS);
        bundleForShortComments.putInt("count", mStoryExtra.getShortComments());
        shortCommentsFragment.setArguments(bundleForShortComments);

        //长评论界面
        CommentsFragment longCommentsFragment = new CommentsFragment();
        Bundle bundleForLongComments = new Bundle();
        bundleForLongComments.putInt("id", id);
        bundleForLongComments.putString("url", UrlConstants.LONG_COMMENTS);
        bundleForShortComments.putInt("counts", mStoryExtra.getLongComments());
        longCommentsFragment.setArguments(bundleForLongComments);

        mFragmentArrayList.add(shortCommentsFragment);
        mFragmentArrayList.add(longCommentsFragment);
        CommentPagerAdapter adapter = new CommentPagerAdapter(getSupportFragmentManager(), mFragmentArrayList, tabList);
        //mViewPager.setOffscreenPageLimit(0);
        mViewPager.setAdapter(adapter);
        mViewPager.addOnPageChangeListener(pageListener);

        mTabLayout.setupWithViewPager(mViewPager);//将TabLayout和ViewPager关联起来。
        mTabLayout.setTabsFromPagerAdapter(adapter);//给Tabs设置适配器
    }

    private void initToolbar(Toolbar toolbar) {
        toolbar.setNavigationIcon(R.mipmap.back);//设置导航栏图标
        toolbar.setTitle("共" + mStoryExtra.getShortComments() + "条");//设置主标题
        toolbar.setNavigationOnClickListener(v -> finish());
    }

    private ViewPager.OnPageChangeListener pageListener = new ViewPager.OnPageChangeListener() {


        @Override
        public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

        }

        @Override
        public void onPageSelected(int position) {

        }

        @Override
        public void onPageScrollStateChanged(int state) {

        }
    };
}

File: app/src/main/java/com/yao/zhihudaily/ui/daily/CommentsActivity.kt
package com.yao.zhihudaily.ui.daily

import android.os.Bundle
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.yao.zhihudaily.R
import com.yao.zhihudaily.model.StoryExtra
import com.yao.zhihudaily.net.UrlConstants
import com.yao.zhihudaily.tool.Constant
import com.yao.zhihudaily.ui.BaseActivity
import kotlinx.android.synthetic.main.activity_comments.*
import java.util.*

/**
 *
 * @author Yao
 * @date 2016/8/30
 */
class CommentsActivity : BaseActivity() {

    private var mStoryExtra: StoryExtra? = null
    private val mFragmentArrayList = ArrayList<Fragment>(2)

    private val pageListener = object : ViewPager.OnPageChangeListener {


        override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {

        }

        override fun onPageSelected(position: Int) {

        }

        override fun onPageScrollStateChanged(state: Int) {

        }
    }


    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_comments)

        val id = intent.getIntExtra(Constant.ID, 0)
        mStoryExtra = intent.getSerializableExtra(Constant.STORY_EXTRA) as StoryExtra

        initToolbar(toolbar)


        val tabList = ArrayList<String>()
        tabList.add("短评论(" + mStoryExtra!!.shortComments + ")")
        tabList.add("长评论(" + mStoryExtra!!.longComments + ")")
        tab_layout!!.addTab(tab_layout!!.newTab().setText(tabList[0]))//添加tab选项卡
        tab_layout!!.addTab(tab_layout!!.newTab().setText(tabList[1]))

        //短评论界面
        val shortCommentsFragment = CommentsFragment()
        val bundleForShortComments = Bundle()
        bundleForShortComments.putInt(Constant.ID, id)
        bundleForShortComments.putString(Constant.URL, UrlConstants.SHORT_COMMENTS)
        bundleForShortComments.putInt(Constant.COUNT, mStoryExtra!!.shortComments)
        shortCommentsFragment.arguments = bundleForShortComments

        //长评论界面
        val longCommentsFragment = CommentsFragment()
        val bundleForLongComments = Bundle()
        bundleForLongComments.putInt(Constant.ID, id)
        bundleForLongComments.putString(Constant.URL, UrlConstants.LONG_COMMENTS)
        bundleForShortComments.putInt(Constant.COUNT, mStoryExtra!!.longComments)
        longCommentsFragment.arguments = bundleForLongComments

        mFragmentArrayList.add(shortCommentsFragment)
        mFragmentArrayList.add(longCommentsFragment)
        val adapter = CommentPagerAdapter(supportFragmentManager, mFragmentArrayList, tabList)
        //view_pager.setOffscreenPageLimit(0);
        view_pager!!.adapter = adapter
        view_pager!!.addOnPageChangeListener(pageListener)

        tab_layout!!.setupWithViewPager(view_pager)//将TabLayout和ViewPager关联起来。
        tab_layout!!.setTabsFromPagerAdapter(adapter)//给Tabs设置适配器
    }

    private fun initToolbar(toolbar: Toolbar) {
        toolbar.setNavigationIcon(R.mipmap.back)//设置导航栏图标
        toolbar.title = "共" + mStoryExtra!!.shortComments + "条"//设置主标题
        toolbar.setNavigationOnClickListener { v -> finish() }
    }
}
